new-lamassu-admin/src/pages/AddMachine/AddMachine.js

```diff
@@ -27,12 +29,40 @@ const SAVE_CONFIG = gql`
     createPairingTotem(name: $name)
   }
 `
+const GET_MACHINES = gql`
+  {
+    machines {
+      name
+      deviceId
+    }
+  }
+`
 
 const useStyles = makeStyles(styles)
 
-const QrCodeComponent = ({ classes, qrCode, close }) => {
-  const [doneButton, setDoneButton] = useState(null)
-  setTimeout(() => setDoneButton(true), 2000)
+const getSize = R.compose(R.length, R.pathOr([], ['machines']))
+
+function usePairDevice(count) {
+  const [on, toggle] = useBoolean(false)
+  const { data, stopPolling, startPolling } = useQuery(GET_MACHINES)
+  const size = getSize(data)
+
+  useLifecycles(() => startPolling(10000), stopPolling)
+  useEffect(() => (size > count ? toggle(true) : undefined))
+
+  return [on]
+}
```

Comment:
it's better if you do a specific query for finding out if the id is paired. 

Issues with this method:
- If someone else adds a machine to the same server this breaks
- If someone removes a machine from this server this breaks